File: 002/solution_002.py
#! /usr/bin/env python


def fibonacci(ceiling):
    a = 0
    b = 1
    fib = a + b

    while fib < ceiling:
        yield fib
        a = b
        b = fib
        fib = a + b


def project_euler_002(n):
    fibs = fibonacci(4000000)
    fibs_even = filter(lambda n: not n % 2, fibs)
    solution = sum(fibs_even)

    print solution


if __name__ == '__main__':
    project_euler_002(4000000)

File: 002/solution_002.js
#! /usr/bin/env node

var project_euler_002 = function() {
    var fibs = [1, 2];
    var fib_last = fibs[fibs.length - 1]
    var i = 1;
    while (true) {
        curr = fibs[i - 1] + fibs[i];
        if (curr >= 4000000) break;
        fib_last = curr;
        fibs.push(fib_last)
        i++
    };
    var fibs_even = fibs.filter((x) => x % 2 == 0);
    var solution = fibs_even.reduce((acc, x, _i, _a) => acc + x, 0);
    console.log("%d", solution);
}

project_euler_002()
